collectors/prometheus.go

```diff
@@ -112,8 +107,16 @@ func (p *PrometheusExporter) registerMetrics() error {
 	metricsMtx.Lock()
 	defer metricsMtx.Unlock()
 
-	for _, collectorFunc := range collectors {
-		err := prometheus.Register(collectorFunc(p.lnd))
+	collectors := []prometheus.Collector{
+		NewChainCollector(p.lnd),
```

Comment:
Why's this preferable to the init functions?

Reply:
There is an initialization parameter now that comes from the config.

Reply:
Needs to be updated now after the merge of the get info collector. The main reason why the existing structure was used was so we'd just need to add the new collector in the file it was created in, rather than here as well. In the end something needs to be updated either way though.

Reply:
Rebased